File: services/matching-engine/internals/types/order.go
package types

import "time"

type Side string
type OrderType string
type Action string
type Role string

const (
	Yes Side = "YES"
	No  Side = "NO"
)

const (
	LIMIT  OrderType = "LIMIT"
	MARKET OrderType = "MARKET"
)

const (
	BUY  Action = "BUY"
	SELL Action = "SELL"
)

const (
	USER  Role = "USER"
	ADMIN Role = "ADMIN"
)

type Order struct {
	OrderId   string
	UserId    string
	MarketId  string
	Symbol    string
	Role      Role
	Price     float64
	Quantity  int
	Filled    int
	Side      Side
	Action    Action
	OrderType OrderType
	Timestamp time.Time
}

File: apps/web-app/src/api/order.ts
import { api } from '@/lib/axios';

export const placeOrder = (
	side: string,
	symbol: string,
	action: string,
	price: number,
	orderType: string,
	quantity: number,
	marketId: string,
) => {
	const endpoint = action.toLowerCase() === 'sell' ? '/order/sell' : '/order/buy';
	return api.post(endpoint, { side, symbol, action, price, orderType, quantity, marketId });
};

export const cancelOrder = (orderId: string, marketId: string) => {
	return api.post('/order/cancel', { orderId, marketId });
};
